hr_holidays_public/models/hr_holidays_public.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2015 2011,2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrHolidaysPublic(models.Model):
+    _name = 'hr.holidays.public'
+    _description = 'Public Holidays'
+    _rec_name = 'year'
+    _order = 'year'
+
+    display_name = fields.Char(
+        'Name',
+        compute='_compute_display_name',
+        store=True,
+    )
+    year = fields.Integer(
+        'Calendar Year',
+        required=True,
+        default=date.today().year
+    )
+    line_ids = fields.One2many(
+        'hr.holidays.public.line',
+        'year_id',
+        'Holiday Dates'
+    )
+    country_id = fields.Many2one(
+        'res.country',
+        'Country'
+    )
+
+    @api.multi
+    @api.constrains('year', 'country_id')
+    def _check_year(self):
+        for line in self:
+            line._check_year_one()
+
+    def _check_year_one(self):
+        if self.country_id:
```

Comment:
Maybe the condition is not necessary. I think you could replace

	...
	    if self.country_id:
            domain = [('year', '=', self.year),
                      ('country_id', '=', self.country_id.id),
                      ('id', '!=', self.id)]
        else:
            domain = [('year', '=', self.year),
                      ('country_id', '=', False),
                      ('id', '!=', self.id)]
	...

with::

	...
		domain = [
			('year', '=', self.year),
			('country_id', '=', self.country_id.id),
			('id', '!=', self.id)
		]
	...

Reply:
👍